lib/package/sns/compileIamRoleToSns.js

```diff
@@ -1,20 +1,22 @@
 'use strict'
 const _ = require('lodash')
 
+const SERVICE_NAME = 'sns'
+
 module.exports = {
   compileIamRoleToSns() {
+    if (!this.shouldCreateDefaultRole(SERVICE_NAME)) {
+      return
+    }
+
     const topicNames = this.getAllServiceProxies()
-      .filter((serviceProxy) => this.getServiceName(serviceProxy) === 'sns')
+      .filter((serviceProxy) => this.getServiceName(serviceProxy) === SERVICE_NAME)
       .map((serviceProxy) => {
         const serviceName = this.getServiceName(serviceProxy)
         const { topicName } = serviceProxy[serviceName]
         return topicName
       })
 
-    if (topicNames.length <= 0) {
```

Comment:
This is taken care of [here](https://github.com/horike37/serverless-apigateway-service-proxy/pull/42/files#diff-50e3aa130a4f97a42ee2cf111c7b1d9dR26)

I actually figured out I need to remove it since I had a drop in test coverage for that specific line

Reply:
👍